### PR TITLE
[glad] Fix Python module import in build process

### LINKED ISSUE #40786: [glad] Build error on x64-windows
Copy issue body from E:/vcpkg/installed/vcpkg/issue_body.md

E:\>vcpkg install glad
Computing installation plan...
The following packages will be built and installed:
    glad[core,loader]:x64-windows@0.1.36
Detecting compiler hash for triplet x64-windows...
-- Automatically setting %HTTP(S)_PROXY% environment variables to "127.0.0.1:7890".
Compiler found: E:/Visiual Studio/Visiual Studio IDE/VC/Tools/MSVC/14.33.31629/bin/Hostx64/x64/cl.exe
Restored 0 package(s) from C:\Users\1\AppData\Local\vcpkg\archives in 99.2 us. Use --debug to see more details.
Installing 1/1 glad[core,loader]:x64-windows@0.1.36...
Building glad[core,loader]:x64-windows@0.1.36...
-- Note: glad only supports static library linkage. Building static library.
-- Using cached Dav1dde-glad-1ecd45775d96f35170458e6b148eb0708967e402.tar.gz.
-- Cleaning sources at E:/vcpkg/buildtrees/glad/src/708967e402-274d415499.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source E:/vcpkg/downloads/Dav1dde-glad-1ecd45775d96f35170458e6b148eb0708967e402.tar.gz
-- Applying patch encoding.patch
-- Applying patch find_python.patch
-- Using source at E:/vcpkg/buildtrees/glad/src/708967e402-274d415499.clean
-- This version of glad uses the compatibility profile. To use the core profile instead, create an overlay port of this with GLAD_PROFILE set to 'core' or set GLAD_PROFILE to 'core' in a custom triplet.
-- This recipe is at E:/vcpkg/ports/glad
-- See the overlay ports documentation at https://github.com/microsoft/vcpkg/blob/master/docs/specifications/ports-overlay.md
-- Found external ninja('1.10.2').
-- Configuring x64-windows
-- Building x64-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: E:/CMake/bin/cmake.exe --build . --config Debug --target install -- -v -j17
    Working Directory: E:/vcpkg/buildtrees/glad/x64-windows-dbg
    See logs for more information:
      E:\vcpkg\buildtrees\glad\install-x64-windows-dbg-out.log

Call Stack (most recent call first):
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_build.cmake:74 (vcpkg_execute_build_process)
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_install.cmake:16 (vcpkg_cmake_build)
  ports/glad/portfile.cmake:143 (vcpkg_cmake_install)
  scripts/ports.cmake:192 (include)

error: building glad:x64-windows failed with: BUILD_FAILED
See https://learn.microsoft.com/vcpkg/troubleshoot/build-failures?WT.mc_id=vcpkg_inproduct_cli for more information.
Elapsed time to handle glad:x64-windows: 3.8 s
Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
  https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+glad
You can submit a new issue at:
  https://github.com/microsoft/vcpkg/issues/new?title=[glad]+Build+error+on+x64-windows&body=Copy+issue+body+from+E%3A%2Fvcpkg%2Finstalled%2Fvcpkg%2Fissue_body.md
```[tasklist]
### Tasks
```

### PR DESCRIPTION
**Pull request by ne0-wu**
Add temporary python._pth file during build to ensure correct module discovery. The file is automatically generated with proper Python version and removed after build completion.

The temporary _pth file ensures the glad module can be imported by adding the source directory to Python's module search path, while maintaining a clean environment by removing the file after build.

This temporarily fixes the glad module import issue when using vcpkg's bundled Python interpreter during build process.

Fixes #40786 

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [ ] SHA512s are updated for each updated download.
- [ ] The "supports" clause reflects platforms that may be fixed by this new version.
- [ ] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [ ] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [ ] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by ne0-wu** [comment]
@microsoft-github-policy-service agree

**Review by FrankXie05 — APPROVED** [review]

**Comment by Osyotr** [comment]
Wouldn't adding this path to `PATH` solve this problem?
https://learn.microsoft.com/en-us/vcpkg/maintainers/functions/vcpkg_cmake_install#add_bin_to_path

**Comment by ne0-wu** [comment]
> Wouldn't adding this path to `PATH` solve this problem? https://learn.microsoft.com/en-us/vcpkg/maintainers/functions/vcpkg_cmake_install#add_bin_to_path

No, it doesn't. According to the original issue (also tested this on my machine), the core issue was sys.path mixing multiple Python environments, causing conflicts with vcpkg's embedded Python during glad library builds.
This is an ugly workaround, but it did temporarily solve the issue.

**Comment by vipcxj** [comment]
same issue in glib #41869
@ne0-wu is there a general solution?

File: ports/glad/portfile.cmake [hunk 1]
```diff
@@ -123,6 +123,21 @@ file(COPY
...
 
+get_filename_component(PYTHON_DIR "${PYTHON3}" DIRECTORY)
+if(PYTHON_DIR MATCHES "vcpkg")
+execute_process(
+    COMMAND "${PYTHON3}" -c "import sys; print(f'python{sys.version_info.major}{sys.version_info.minor}')"
+    OUTPUT_VARIABLE PYTHON_VERSION_NAME
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+)
+file(TO_NATIVE_PATH "${SOURCE_PATH}" GLAD_SOURCE_PATH)
+file(WRITE "${PYTHON_DIR}/python._pth"
```
Comment (BillyONeal) [inline ports/glad/portfile.cmake:134]: This is going to try to write to, for example `%ProgramFiles%`, which will fail. I don't think you can write to the python installation like this. You might be able to ask python go make a venv for you and edit that?

**Comment by BillyONeal** [comment]
@ras0219-msft @vicroms @JavierMatosD and I discussed this today and agree that attempting to write to the python installation directory is unworkable. You might be able to solve this problem by copying the python installation to a writable location, or using virtual environments. If you still need modifications to vcpkg like that, we would be happy to see another PR with changes to that effect.

Thanks for your contribution to vcpkg!

**Pull request closed without merging**